apps/dashboard/app/javascript/packs/prefill_templates/prefill_submit.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+const selectorID = "modal_input_template_name";
+const newNameID = "modal_input_template_new_name";
+
+export function prefillSubmitHandler() {
+  const form = $("#new_batch_connect_session_context");
+
+  const chooseTemplateName = $("#chooseTemplateName");
+  if (chooseTemplateName.length === 0) {
+    return;
+  }
+
+  const chooseTemplateNameError = $("#chooseTemplateNameError")
+  let url;
+
+  $(`#${selectorID}`).on("change", function () {
+    const newName = $(`#${newNameID}`);
+    if ($(this).val() === "") {
+      newName.show();
+    } else {
+      newName.hide();
+    }
+  });
+
+  $("#chooseTemplateNameConfirm").on("click", function () {
+    const name = $(`#${selectorID}`).val() || $(`#${newNameID}`).val();
+    if (name === "") {
+      chooseTemplateNameError.modal('show');
+      return;
+    }
+    chooseTemplateNameError.modal('hide');
+    url.searchParams.set('template', name);
+    form.attr("action", url.href);
+    chooseTemplateName.modal('hide');
+  });
+
+  form.one("submit", function (event) {
```

Comment:
I think I figured out what this javascript is doing. However - you have a bug here. Should be `on` not `one` (somehow `one` doesn't throw errors?).

In any case - I'd ask that we get rid of it anyhow and refactor to have this behavior:
* when the user clicks the 'save template' checkbox - that's when the modal is shown.

I don't think we should get into the business of setting the form's action attr or doing anything with form submission behavior. The modal can set the form text field and the form can just submit with default behavior without javascript interception.

Reply:
Okay, will work on those changes tomorrow.

Reply:
<img width="475" alt="image" src="https://github.com/OSC/ondemand/assets/43995067/54dc338c-bafd-4691-93d9-4b59e8fc2f32">

Updated. Now we display the non-editable name to the user as we are using this user interaction. I think it would be better to use a hidden field so that user can understand what had happened.